airflow/utils/sqlalchemy.py

```diff
@@ -341,7 +341,7 @@ def with_row_locks(
         kwargs["nowait"] = True
     if skip_locked:
         kwargs["skip_locked"] = True
-    return query.with_for_update(**kwargs)
+    return query.with_for_update(**kwargs, key_share=True)
```

Comment:
Should this be configurable in the function? (i.e. add this as an argument and defaulting to `True`)

Reply:
it's not really needed at this time AFAIK but yeah that would be a reasonable thing to do :)

Reply:
done btw